server/xcode/xcode_darwin.go

```diff
@@ -90,6 +96,14 @@ func (x *xcodeLocator) locate() error {
 	urlRefs := make([]C.CFURLRef, n)
 	C.CFArrayGetValues(urlsArrayRef, C.CFRange{0, n}, (*unsafe.Pointer)(unsafe.Pointer(&urlRefs[0])))
 
+	var defaultDeveloperDir string
+	out, err := exec.Command("xcode-select", "-p").Output()
+	if err != nil {
+		log.Warningf("Error calling xcode-select: %s", err.Error())
+	} else {
+		defaultDeveloperDir = strings.TrimSpace(string(out))
+	}
+
 	versionMap := make(map[string]*xcodeVersion)
 	for _, urlRef := range urlRefs {
 		path := "/" + strings.TrimLeft(stringFromCFString(C.CFURLGetString(C.CFURLRef(urlRef))), filePrefix)
```

Comment:
This loop would be a lot clearer if we:
 - defined the "xcodePlist" struct up at the top of this file
 - pulled the plist parsing out into a function that takes a path (string) and returns a xcodePlist struct and error

Reply:
Addressed in #1354.